### PR TITLE
Fix data parameter always undefined on callback for track method

### PR DESCRIPTION
**Pull request by albertmourato**
Fixes: https://github.com/segmentio/analytics-node/issues/230

### CODE REVIEW

**Comment by albertmourato** [comment]
@danieljackins I didn't need to change anywhere else since the place I added solved my problem. Feel free to add in the other places.

**Comment by albertmourato** [comment]
@danieljackins just added the callback parameter to the `this.flush` call as you asked. Could you please review this PR again?
There are other people facing the same problem.
Check https://github.com/segmentio/analytics-node/issues/230

File: index.js [hunk 1]
```diff
@@ -205,12 +205,12 @@ class Analytics {
...
+      this.flush(callback)
       return
     }
 
     if (this.queue.length >= this.flushAt) {
-      this.flush()
+      this.flush(callback)
     }
 
     if (this.flushInterval && !this.timer) {
```
Comment (pbassut) [inline index.js:216]: There's also another one here:
https://github.com/segmentio/analytics-node/blob/d2c565d27110b8c3846372a8d8aacc69b0ef83ca/index.js#L217

**Review by pbassut — CHANGES_REQUESTED** [review]
Mind make that little change? Just so we're consistent on all three callback

File: index.js (hunk 1 above)
Comment (albertmourato) [reply index.js:216]: thanks, @pbassut
just applied the suggestion!

**Review by nd4p90x — APPROVED** [review]

File: index.js [hunk 2]
```diff
@@ -213,16 +213,16 @@ class Analytics {
...
     if (!this.flushed) {
       this.flushed = true
-      this.flush()
+      this.flush(callback)
       return
     }
 
     if (this.queue.length >= this.flushAt) {
-      this.flush()
+      this.flush(callback)
```
Comment (Dahaden) [inline index.js:221]: Weird bug, may not cause issues, but this actually causes the callback to be called twice now as its the exact same callback pulled from the item when flushed.
https://github.com/segmentio/analytics-node/blob/master/index.js#L247-L259

Comment (pooyaj) [reply index.js:221]: cc - @pbassut

Comment (BrandonNoad) [reply index.js:221]: Yes, the callback is running twice for me as well. Is there a plan to fix that? Or should I log a new issue?

Comment (pbassut) [reply index.js:221]: We should have a fix ASAP

Comment (BrandonNoad) [reply index.js:221]: Hi. Was the fix for this released with v5.1.0?

Comment (pooyaj) [reply index.js:221]: @pbassut could you please check if this is already fixed?

Comment (pbassut) [reply index.js:221]: @pooyaj It's fixed locally. I'm writing some tests and I'll create a PR for it.

Comment (pbassut) [reply index.js:221]: @BrandonNoad Can you give me a minimal working example that illustrates the issue? 

For issue `#1`: seems unlikely given they're only called when the request to segment servers actually finishes(successfully or not).

As for issue `#2`: I'll have to see some code.

Comment (pbassut) [reply index.js:221]: @BrandonNoad Please open a new issue for it though.

Comment (BrandonNoad) [reply index.js:221]: @pbassut Yes, I can open a new issue with some code to reproduce the behaviour.

> For issue #1: seems unlikely given they're only called when the request to segment servers actually finishes(successfully or not).

I may have been incorrect in saying "too early". That was just an assumption based on the fact that the callback is being called twice and in the first call, `batch` is `undefined`.

Comment (BrandonNoad) [reply index.js:221]: New issue is here: https://github.com/segmentio/analytics-node/issues/300

**Pull request merged**